x-pack/plugins/observability_solution/apm/public/components/alerting/ui_components/alert_details_app_section/view_in_apm_button.tsx

```diff
@@ -28,7 +29,11 @@ export function ViewInAPMButton({
   to: string;
   kuery?: string;
 }) {
-  const { share } = useApmPluginContext() || {};
+  // this component is rendered in the Observability Plugin, so we can use the ObservabilityPublicPluginsStart type
+  const {
+    services: { share },
+  } = useKibana<ObservabilityPublicPluginsStart>();
```

Comment:
This is confusing but we're already doing the same in `failed_transaction_chart.tsx`, `x-latency_chart.tsx` and `throughput_chart.tsx`